src/Tools/_framework/Paths/PublicEditor.jsx

```diff
@@ -470,16 +489,135 @@ export function PublicEditor() {
                   borderBottom="solid 1px"
                   borderColor="doenet.mediumGray"
                 >
-                  <CodeMirror
-                    editorRef={editorRef}
-                    setInternalValueTo={doenetML}
-                    onBeforeChange={(value) => {
-                      textEditorDoenetML.current = value;
-                      if (!codeChanged) {
-                        setCodeChanged(true);
-                      }
-                    }}
-                  />
+                  <Box
```

Comment:
I was going to make all the same comments in this file as in PortfolioActivityEditor.jsx.  Can we consolidate the duplicate code into a utility function that we can call in both places?

Reply:
Now consolidated into ` <ErrorWarningPopovers>`, thanks!